### PR TITLE
`Assessment`: Add possibility to sort complaints

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

A large course can accumulate more than 200 complaints on a single exercise. These are presented in a big table. To help keep a good overview, the option to sort these complaints

### Description

This PR add the options to sort the tables in the exercise assessment dashboards for manually assessed submissions, complaints and more feedback requests.
The idea in general is that tutors can now access the submissions that still need work easier (sorted to the top) and submissions that have waited longer are prioritized (again sorted to the top).
In particular the submissions are now sorted by default based on their submission date with older submissions being on top. And the option to start a new assessment is the top most row. Complaints and more feedback requests are sorted by default based on their possible action, the ones that still need assessment are at the top with the prioritization of those being submitted earlier.
All other columns can also be used for sorting by clicking on the name of the column in the header of the table.

Additionally the list of complaints is updated to also show the response time of complaints.

### Steps for Testing

Prerequisites:
- 1 instructor
- Multiple (>= 4) students
- 1 normal exercise
- 1 exam exercise with two assessment rounds

1. Create the exercises
2. Participate as the students
3. Assess some but not all submissions and complain or request more feedback as the student
4. Check that all tables are sorted correctly by default and that all other sorting options work
5. Check that the list of complaints (accessible via the statistics on the top right of the assessment dashboard) is displayed correctly

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

sort.service.ts: 93.93% | 94.44% (before: 85.29% | 91.66%)

### Screenshots

Assessment Dashboard:
Bildschirmfoto 2022-02-06 um 19 54 22
Bildschirmfoto 2022-02-06 um 19 53 56

List of complaints:
Bildschirmfoto 2022-02-06 um 19 53 28

### CODE REVIEW

**Review by mallgaier — APPROVED** [review]
Tested with @lBruchmann , @Strohgelaender and @VinnavanSembian on TS1.
Works as expected.

**Review by VinnavanSembian — DISMISSED** [review]
tested on ts1 with @Strohgelaender @mallgaier @lBruchmann. Everything works as expected!

**Review by lBruchmann — DISMISSED** [review]
Tested on ts1. Tested together with @mallgaier, @VinnavanSembian  and @Strohgelaender. Everything works as expected. Great feature!

File: src/main/webapp/app/shared/service/sort.service.ts [hunk 1]
```diff
@@ -7,54 +7,72 @@ import dayjs from 'dayjs/esm';
...
             }
 
-            return SortService.compareBasic(valueA, valueB, asc);
+            return compareValue;
         });
     }
 
     private static compareWithUndefinedNull(valueA: any, valueB: any) {
-        if (!valueA && !valueB) {
+        if ((valueA === null || valueA === undefined) && (valueB === null || valueB === undefined)) {
```
Comment (Strohgelaender) [inline src/main/webapp/app/shared/service/sort.service.ts:50]:
```suggestion
        if (valueA == null && valueB == null ) {
```
You can use `==` here and below to check against both `null` and `undefined`.

**Review by Strohgelaender — DISMISSED** [review]
Tested on TS1, works as expected 👍

File: src/main/webapp/app/shared/service/sort.service.ts (hunk 1 above)
Comment (JohannesStoehr) [reply src/main/webapp/app/shared/service/sort.service.ts:50]: I tried that. The linter did not like it :(

File: src/main/webapp/app/complaints/complaint.service.ts [hunk 2]
```diff
@@ -181,6 +181,39 @@ export class ComplaintService implements IComplaintService {
...
+     * @return returns the passed time in seconds
+     */
+    getResponseTimeInSeconds(complaint: Complaint): number {
+        let responseTime;
+        if (complaint.accepted !== undefined) {
+            responseTime = complaint.complaintResponse?.submittedTime?.diff(complaint.submittedTime, 'seconds') || NaN;
+        } else {
+            responseTime = dayjs().diff(complaint.submittedTime, 'seconds');
+        }
+        return responseTime;
```
Comment (Strohgelaender) [inline src/main/webapp/app/complaints/complaint.service.ts:196]:
```suggestion
        if (complaint.accepted !== undefined) {
            return complaint.complaintResponse?.submittedTime?.diff(complaint.submittedTime, 'seconds') || NaN;
        } else {
            return dayjs().diff(complaint.submittedTime, 'seconds');
        }
```

**Review by Strohgelaender — APPROVED** [review]

**Review by lBruchmann — APPROVED** [review]

**Review by VinnavanSembian — APPROVED** [review]

**Review by mallgaier — APPROVED** [review]

**Pull request merged**